tests/fail/tree_borrows/unique.uniq.stderr

```diff
@@ -0,0 +1,38 @@
+error: Undefined Behavior: write access through <TAG> is forbidden
+  --> $DIR/unique.rs:LL:CC
+   |
+LL |         *uniq.as_ptr() = 2;
+   |         ^^^^^^^^^^^^^^^^^^ write access through <TAG> is forbidden
+   |
+   = help: this indicates a potential bug in the program: it performed an invalid operation, but the Tree Borrows rules it violated are still experimental
+   = help: the accessed tag <TAG> is a child of the conflicting tag <TAG>
+   = help: the conflicting tag <TAG> has state Frozen which forbids this child write access
+help: the accessed tag <TAG> was created here
+  --> $DIR/unique.rs:LL:CC
+   |
+LL |     let uniq = Unique::new(rawptr).unwrap();
+   |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+help: the conflicting tag <TAG> was created here, in the initial state Reserved
+  --> $DIR/unique.rs:LL:CC
+   |
+LL |     let uniq = Unique::new(rawptr).unwrap();
+   |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Comment:
This looks a bit strange, the accessed and conflict tag seem to be the same?

Is this a new thing with unique or an existing diagnostics issue?

Reply:
These are in fact not the same, there's a special case in the diagnostics when they are (the line "`= help: the accessed tag <TAG> is a child of the conflicting tag <TAG>`" would not be printed if they were equal).
Those are two different tags, but the `Span` for the location where they were created is the same.

Reply:
Ah, I guess there are two retags here, due to the `unwrap`...

Maybe use `new_unchecked` instead? That doesn't solve the general problem (when there are many conflicting pointers, maybe we shouldn't always report the closest like here), but could at least help for this test.

Reply:
There might be a bug somewhere else in the retagging, `new_unchecked` doesn't seem to do *any* retagging.

Reply:
If I use `new_unchecked` then there's no difference in this test between having `-Zmiri-unique-is-unique` or not.

Reply:
Ah, that's a problem.

You might have to adjust this function: https://github.com/rust-lang/rust/blob/08384b854f4bcbba9459f22665063e7c45877dfe/compiler/rustc_mir_transform/src/add_retag.rs#L13-L15.